xml_callbacks.go

```diff
@@ -0,0 +1,42 @@
+package libxml2
+
+import "C"
+import (
+	"bytes"
+	"github.com/lestrrat-go/libxml2/clib"
+	"io"
+	"io/ioutil"
+)
+
+type inMemoryCallback struct {
+	uri         string
+	data        []byte
+}
+
+func (i *inMemoryCallback) Open(_ string) (io.ReadCloser, error) {
+	return ioutil.NopCloser(bytes.NewReader(i.data)), nil
+}
+
+func (i *inMemoryCallback) Match(uri string) bool {
+	return i.uri == uri
+}
+
+var _ clib.Callback = (*inMemoryCallback)(nil)
+
+func InMemoryCallback(xmlURI string, data []byte) clib.Callback {
+	return &inMemoryCallback{
+		uri:  xmlURI,
+		data: data,
+	}
+}
+
+func RegisterInputCallback(callbacks ...clib.Callback) {
```

Comment:
What happens when you have multiple callbacks? Is it legal to do so? 

I guess it is... if you want to register multiple things to be inserted, you certainly need to do that. But what if you use the same matcher? Probably needs documentation.

```go
libxml2.RegisterInputCallback(
    libxml2.FooBarCallback("one"),
    libxml2.FooBarCallback("one"), // what happens when there's a duplicate?
 )
 ```

Reply:
I guess you also need to document what happens when you call `RegisterInputCallback` multiple times

Reply:
Other things to include:
* You can only do exact matching
* You can only insert the content verbatim
* Because of underlying C implementation, the callback list is global. Users are responsible for syncing which callbacks are registered.